CONTRIBUTING.rst

```diff
@@ -19,8 +22,48 @@ Contributor Guide
 
    * Now, you have remote repositories named:
 
-      - ``upstream``, which refers to the ``networkx`` repository
-      - ``origin``, which refers to your personal fork
+     - ``upstream``, which refers to the ``networkx`` repository
+     - ``origin``, which refers to your personal fork
+
+   * Next, you need to set up your build environment.
+     Here are instructions for two popular environment managers:
+   
+     * ``venv`` (pip based)
+     
+       ::
+     
+         # Create a virtualenv named ``networkx-dev`` that lives in the directory of
+         # the same name
+         python -m venv networkx-dev
+         # Activate it
+         source networkx-dev/bin/activate
+         # Install all development and runtime dependencies of networkx
+         pip install -r <(cat requirements/*.txt)
+         # Build and install networkx from source
+         pip install -e .
+         # Test your installation
+         PYTHONPATH=. pytest networkx
+     
+     * ``conda`` (Anaconda or Miniconda)
+    
+       ::
+ 
+         # Create a conda environment named ``networkx-dev``
+         conda create --name networkx-dev
+         # Activate it
+         conda activate networkx-dev
+         # Install major development and runtime dependencies of networkx
+         # (the rest can be installed from conda-forge or pip, if needed)
+         conda install `for i in requirements/{default,test,doc,developer,extras}.txt; do echo -n " --file $i "; done`
```

Comment:
`optional` is missing from this list.

Also, I don't use conda often, but is there no better way to install from requirements files? I know you can use pip from within conda, but maybe that usage is frowned upon.

Reply:
I am not familiar with conda and not sure how to use it.  I would be fine with just removing it, since I don't use it and don't know if this recipe even works.  Also it will likely cause problems with pygraphviz due to a buggy version of graphviz in conda for the last two years.  I don't know if the other packages work or not.  But I believe a lot of new users are steered toward conda for some reason, so it may be worth keeping this around for them.  Maybe someone else on the core team has experience working with conda.

Reply:
I tried the recipe as-is and got:

```
PackagesNotFoundError: The following packages are not available from current channels:                                       

  - nb2plots[version='>=0.6']                                                                                                
  - texext[version='>=0.6.6']                                                                                                
  - sphinx-gallery==0.7.0                                                                                                    
  - pre-commit                                                                                                               
  - sphinx_rtd_theme==0.5.0                                                                                                  

Current channels:                                                                                                            

  - https://repo.anaconda.com/pkgs/main/linux-64                                                                             
  - https://repo.anaconda.com/pkgs/main/noarch                                                                               
  - https://repo.anaconda.com/pkgs/r/linux-64                                                                                
  - https://repo.anaconda.com/pkgs/r/noarch
```

Using the conda forge channel (i.e. add `-c conda-forge`) fixed this problem for me, though I don't know whether that's considered a good/stable practice.

Reply:
I use conda but I'm no expert. I would include the requirement files like:
```
conda install -c conda-forge --file default.txt --file test.txt --file doc.txt --file developer.txt --file extras.txt
```
I get better results using ```-c conda-forge``` than from the default channel (like Ross did).

The ```conda-forge``` channel has graphviz-2.42 so, e.g. for travis, if we use ```conda install -c conda-forge graphviz=2.42``` that should work.  I'm not sure, but think that pygraphviz v1.5 requires graphviz v2.38 in conda-forge.  pygraphviz v1.6 should fix that... (or maybe there is a way to have conda-forge require >=2.38 but not 2.40)  I don't know the syntax possibilities for version spec in requirements files.